.doom.d/config.org
#+TITLE: My emacs doom configuration
#+AUTHOR: zhou mingjun
#+EMAIL: zhoumingjun@gmail.com
#+PROPERTY: header-args:emacs-lisp :tangle yes :cache yes :padline no :results silent


* TOC :TOC_3_gh:noexport:
- [[#configuration][Configuration]]
  - [[#personal-information][Personal Information]]
    - [[#set-full-name][Set full name]]
    - [[#set-email-address][Set email address]]
  - [[#font][Font]]
  - [[#theme][Theme]]
  - [[#org][Org]]
    - [[#set-org-directory][Set org-directory]]
    - [[#org-plantuml][org plantuml]]
    - [[#org-latex][org latex]]
    - [[#org-export][org export]]
    - [[#org-crypt][org crypt]]
    - [[#org-roam][org roam]]
  - [[#line-number][Line-number]]
  - [[#which-key][which-key]]
  - [[#tabs][tabs]]
  - [[#doom][doom]]
    - [[#dashboard][dashboard]]
  - [[#tramp][Tramp]]
  - [[#latex][latex]]
  - [[#assembly][assembly]]

* Configuration
** Personal Information
*** Set full name
#+BEGIN_SRC emacs-lisp
(setq user-full-name "zhou mingjun")
#+END_SRC
*** Set email address
#+BEGIN_SRC emacs-lisp
(setq user-mail-address "zhoumingjun@gmail.com")
#+END_SRC
** Font
#+BEGIN_SRC emacs-lisp
(setq doom-font (font-spec :family "JetBrainsMono Nerd Font Mono" :size 12)
      doom-variable-pitch-font (font-spec :family "sans"))
#+END_SRC
** Theme
#+BEGIN_SRC emacs-lisp
(setq doom-theme 'doom-dracula) ;; << This line enables the theme
#+END_SRC
** Org
*** Set org-directory
#+BEGIN_SRC emacs-lisp
(after! org
  (setq org-log-done 'note
        org-log-into-drawer t)

  ;; set org file directory
  (setq org-files-directory "~/org/")

  ;; set agenda files
  ;; (setq org-agenda-files (list org-files-directory))

  ;; (setq org-agenda-files
  ;;   (apply 'append
  ;;     (mapcar
  ;;       (lambda (directory)
  ;;         (directory-files-recursively
  ;;           directory org-agenda-file-regexp))
  ;;             '("~/plan" ))))

  ;; Set different bullets, with one getting a terminal fallback.
  (setq org-superstar-headline-bullets-list
      '("☯" "☰" "☱" "☲" "☳" "☴" "☵" "☶" "☷" "☷" "☷" "☷"))

  ;; set org
  (setq org-latex-default-packages-alist '(("" "xeCJK" t  ("xelatex")) ("AUTO" "inputenc" t  ("pdflatex")) ("T1" "fontenc" t  ("pdflatex")) ("" "graphicx" t) ("" "grffile" t) ("" "longtable" nil) ("" "wrapfig" nil) ("" "rotating" nil) ("normalem" "ulem" t) ("" "amsmath" t) ("" "textcomp" t) ("" "amssymb" t) ("" "capt-of" nil) ("hidelinks" "hyperref" nil))
))
  ;; (add-to-list 'org-latex-default-packages-alist '("" "xeCJK" t ("xelatex"))))

(use-package! org-projectile
   :after org
   :config
   (progn
     (org-projectile-per-project) ; strategy to use a separate .org file within each project
     (setq
      org-projectile-per-project-filepath "todo.org" ; use an .org file named <project>.org for each project
      org-agenda-files (append org-agenda-files (org-projectile-todo-files)) ; get .org file for all known projects & add to `org-agenda-files'
      )))

(use-package! ox-twbs)
(use-package ox-latex
  :defer t
  :init
  (setq org-latex-compiler "xelatex"))

(setq deft-directory "~/org/"
      deft-recursive t)
#+END_SRC
*** org plantuml
#+begin_src emacs-lisp
(setq plantuml-default-exec-mode 'jar)
(setq plantuml-java-args (list "-Dhttp.proxyHost=127.0.0.1" "-Dhttp.proxyPort=1081" "-Dhttps.proxyHost=127.0.0.1" "-Dhttps.proxyPort=1081" "-Djava.awt.headless=true" "-jar" "--illegal-access=deny"))
#+end_src

*** org latex
#+begin_src emacs-lisp
(setq org-latex-toc-command "\\tableofcontents \\clearpage")
#+end_src
*** org export
#+begin_src emacs-lisp
(setq org-export-allow-bind-keywords t)
(setq org-html-table-default-attributes
      '(:class "table table-striped" ))
#+end_src
*** org crypt
#+begin_src emacs-lisp
(setq org-crypt-key "infraservice")
#+end_src

*** org roam
#+begin_src emacs-lisp
(use-package! websocket
    :after org-roam)

(use-package! org-roam-ui
    :after org-roam ;; or :after org
;;  :hook
;;         normally we'd recommend hooking orui after org-roam, but since org-roam does not have
;;         a hookable mode anymore, you're advised to pick something yourself
;;         if you don't care about startup time, use
    :hook (after-init . org-roam-ui-mode)
    :config
    (setq org-roam-ui-sync-theme t
          org-roam-ui-follow t
          org-roam-ui-update-on-save t
          org-roam-ui-open-on-start t))
#+end_src
** Line-number
#+BEGIN_SRC emacs-lisp
(setq display-line-numbers-type t)
#+END_SRC
** which-key
#+BEGIN_SRC emacs-lisp
(setq which-key-idle-delay 0.01)
#+END_SRC
** tabs
disable this
#+BEGIN_SRC emacs-lisp :exports none
(after! centaur-tabs
  (centaur-tabs-group-by-projectile-project))
#+END_SRC
** doom
*** dashboard
Setup Doom dashboard
logo from https://commons.wikimedia.org/wiki/File:EmacsIcon.svg
#+BEGIN_SRC emacs-lisp
(setq +doom-dashboard-banner-padding '(0 . 2)
      +doom-dashboard-menu-sections
  '(("Reload last session"
     :icon (all-the-icons-octicon "history" :face 'doom-dashboard-menu-title)
     :when (cond ((require 'persp-mode nil t)
                 (file-exists-p (expand-file-name persp-auto-save-fname persp-save-dir)))
                 ((require 'desktop nil t)
                  (file-exists-p (desktop-full-file-name))))
     :face (:inherit (doom-dashboard-menu-title bold))
     :action doom/quickload-session)
    ("Open org-agenda"
     :icon (all-the-icons-octicon "calendar" :face 'doom-dashboard-menu-title)
     :when (fboundp 'org-agenda)
     :action org-agenda)
    ("Recently opened files"
     :icon (all-the-icons-octicon "file-text" :face 'doom-dashboard-menu-title)
     :action recentf-open-files)
    ("Open project"
     :icon (all-the-icons-octicon "briefcase" :face 'doom-dashboard-menu-title)
     :action projectile-switch-project)
    ("Jump to bookmark"
     :icon (all-the-icons-octicon "bookmark" :face 'doom-dashboard-menu-title)
     :action bookmark-jump)
    ("Open private configuration"
     :icon (all-the-icons-octicon "tools" :face 'doom-dashboard-menu-title)
     :when (file-directory-p doom-private-dir)
     :action doom/open-private-config)
    ("Open documentation"
     :icon (all-the-icons-octicon "book" :face 'doom-dashboard-menu-title)
     :action doom/help))
      +doom-dashboard-banner-file "emacs.svg"
      +doom-dashboard-banner-dir "~/.doom.d/"
)
#+END_SRC

** Tramp
#+BEGIN_SRC emacs-lisp
(use-package! tramp
  :custom
  (tramp-use-ssh-controlmaster-options nil) ; Don't override SSH config.
  (tramp-default-method "ssh")    ; ssh is faster than scp and supports ports.
  (tramp-password-prompt-regexp   ; Add verification code support.
   (concat
    "^.*"
    (regexp-opt
     '("passphrase" "Passphrase"
       "password" "Password"
       "Verification code"
       "Login Ip:")
     t)
    ".*:\0? *")))

#+END_SRC


** latex
#+BEGIN_SRC emacs-lisp
(add-hook! LaTeX-mode-hook
  (add-to-list TeX-command-list '("XeLaTeX" "%`xelatex%(mode)%' %t" TeX-run-TeX nil t))
  (setq TeX-command-default "XeLaTeX"
        TeX-save-query nil
        TeX-show-compilation t)
  (setq pdf-latex-command "xelatex"))

#+END_SRC
** assembly
#+begin_src emacs-lisp
 (setq  x86-lookup-pdf "~/Documents/325383-sdm-vol-2abcd.pdf")

 (global-set-key (kbd "C-h x") #'x86-lookup)
#+end_src


.doom.d/config.el
(setq user-full-name "zhou mingjun")
(setq user-mail-address "zhoumingjun@gmail.com")
(setq doom-font (font-spec :family "JetBrainsMono Nerd Font Mono" :size 12)
      doom-variable-pitch-font (font-spec :family "sans"))
(setq doom-theme 'doom-dracula) ;; << This line enables the theme
(after! org
  (setq org-log-done 'note
        org-log-into-drawer t)

  ;; set org file directory
  (setq org-files-directory "~/org/")

  ;; set agenda files
  ;; (setq org-agenda-files (list org-files-directory))

  (setq org-agenda-files
    (apply 'append
      (mapcar
        (lambda (directory)
          (directory-files-recursively
            directory org-agenda-file-regexp))
              '("~/plan" ))))

  ;; Set different bullets, with one getting a terminal fallback.
  (setq org-superstar-headline-bullets-list
      '("☯" "☰" "☱" "☲" "☳" "☴" "☵" "☶" "☷" "☷" "☷" "☷"))

  ;; set org
  (setq org-latex-default-packages-alist '(("" "xeCJK" t  ("xelatex")) ("AUTO" "inputenc" t  ("pdflatex")) ("T1" "fontenc" t  ("pdflatex")) ("" "graphicx" t) ("" "grffile" t) ("" "longtable" nil) ("" "wrapfig" nil) ("" "rotating" nil) ("normalem" "ulem" t) ("" "amsmath" t) ("" "textcomp" t) ("" "amssymb" t) ("" "capt-of" nil) ("hidelinks" "hyperref" nil))
))
  ;; (add-to-list 'org-latex-default-packages-alist '("" "xeCJK" t ("xelatex"))))

(use-package! org-projectile
   :after org
   :config
   (progn
     (org-projectile-per-project) ; strategy to use a separate .org file within each project
     (setq
      org-projectile-per-project-filepath "todo.org" ; use an .org file named <project>.org for each project
      org-agenda-files (append org-agenda-files (org-projectile-todo-files)) ; get .org file for all known projects & add to `org-agenda-files'
      )))

(use-package! ox-twbs)
(use-package ox-latex
  :defer t
  :init
  (setq org-latex-compiler "xelatex"))

(setq deft-directory "~/org/"
      deft-recursive t)
(setq plantuml-default-exec-mode 'jar)
(setq plantuml-java-args (list "-Dhttp.proxyHost=127.0.0.1" "-Dhttp.proxyPort=1081" "-Dhttps.proxyHost=127.0.0.1" "-Dhttps.proxyPort=1081" "-Djava.awt.headless=true" "-jar" "--illegal-access=deny"))
(setq org-latex-toc-command "\\tableofcontents \\clearpage")
(setq org-export-allow-bind-keywords t)
(setq org-html-table-default-attributes
      '(:class "table table-striped" ))
(setq org-crypt-key "infraservice")
(use-package! websocket
    :after org-roam)

(use-package! org-roam-ui
    :after org-roam ;; or :after org
;;  :hook
;;         normally we'd recommend hooking orui after org-roam, but since org-roam does not have
;;         a hookable mode anymore, you're advised to pick something yourself
;;         if you don't care about startup time, use
    :hook (after-init . org-roam-ui-mode)
    :config
    (setq org-roam-ui-sync-theme t
          org-roam-ui-follow t
          org-roam-ui-update-on-save t
          org-roam-ui-open-on-start t))
(setq display-line-numbers-type t)
(setq which-key-idle-delay 0.01)
(after! centaur-tabs
  (centaur-tabs-group-by-projectile-project))
(setq +doom-dashboard-banner-padding '(0 . 2)
      +doom-dashboard-menu-sections
  '(("Reload last session"
     :icon (all-the-icons-octicon "history" :face 'doom-dashboard-menu-title)
     :when (cond ((require 'persp-mode nil t)
                 (file-exists-p (expand-file-name persp-auto-save-fname persp-save-dir)))
                 ((require 'desktop nil t)
                  (file-exists-p (desktop-full-file-name))))
     :face (:inherit (doom-dashboard-menu-title bold))
     :action doom/quickload-session)
    ("Open org-agenda"
     :icon (all-the-icons-octicon "calendar" :face 'doom-dashboard-menu-title)
     :when (fboundp 'org-agenda)
     :action org-agenda)
    ("Recently opened files"
     :icon (all-the-icons-octicon "file-text" :face 'doom-dashboard-menu-title)
     :action recentf-open-files)
    ("Open project"
     :icon (all-the-icons-octicon "briefcase" :face 'doom-dashboard-menu-title)
     :action projectile-switch-project)
    ("Jump to bookmark"
     :icon (all-the-icons-octicon "bookmark" :face 'doom-dashboard-menu-title)
     :action bookmark-jump)
    ("Open private configuration"
     :icon (all-the-icons-octicon "tools" :face 'doom-dashboard-menu-title)
     :when (file-directory-p doom-private-dir)
     :action doom/open-private-config)
    ("Open documentation"
     :icon (all-the-icons-octicon "book" :face 'doom-dashboard-menu-title)
     :action doom/help))
      +doom-dashboard-banner-file "emacs.svg"
      +doom-dashboard-banner-dir "~/.doom.d/"
)
(use-package! tramp
  :custom
  (tramp-use-ssh-controlmaster-options nil) ; Don't override SSH config.
  (tramp-default-method "ssh")    ; ssh is faster than scp and supports ports.
  (tramp-password-prompt-regexp   ; Add verification code support.
   (concat
    "^.*"
    (regexp-opt
     '("passphrase" "Passphrase"
       "password" "Password"
       "Verification code"
       "Login Ip:")
     t)
    ".*:\0? *")))
(add-hook! LaTeX-mode-hook
  (add-to-list TeX-command-list '("XeLaTeX" "%`xelatex%(mode)%' %t" TeX-run-TeX nil t))
  (setq TeX-command-default "XeLaTeX"
        TeX-save-query nil
        TeX-show-compilation t)
  (setq pdf-latex-command "xelatex"))
(setq  x86-lookup-pdf "~/Documents/325383-sdm-vol-2abcd.pdf")

(global-set-key (kbd "C-h x") #'x86-lookup)
